webapp/src/js/services/geolocation.js

```diff
@@ -1,12 +1,17 @@
 angular.module('inboxServices').service('Geolocation',
   function(
-    $q
+    $q,
+    $window
   ) {
     'use strict';
     'ngInject';
 
     return function() {
 
+      if ($window.medicmobile_android && typeof $window.medicmobile_android.getLocationPermissions === 'function') {
```

Comment:
Optional nit: wrapping this in a try/catch block. We expect Geolocation service to return a promise, and I think that, if this call throws, forms will just never open.